posthog/api/async_migration.py

```diff
@@ -124,6 +140,9 @@ def resume(self, request, **kwargs):
 
     def _force_stop(self, rollback: bool):
         migration_instance = self.get_object()
+        if migration_instance.status == MigrationStatus.Starting:
+            if async_migration_from_starting_to_rolled_back_atomic(migration_instance):
```

Comment:
Don't put code that has side-effects into conditionals. In general prefer functional coding practices where you separate calculations from side-effects.